### ISSUE TITLE
bpm plugin results not reflected in file metadata

### ISSUE THREAD

**Issue by jfunction** [beetbox/beets#1991]
### Problem

After using the bpm plugin, the results are stored in the beets library but not in the files themselves. I couldn't find any information about why this should be the case. 

I have recently installed beets and tried the bpm plugin. I imported some of my music which was copied over to `~/Music/Radiohead` 

My config file originally just contained the reference to the plugin, after which I typed `beet bpm Prove Yourself` at the cmd line. The resulting bpm was added to the song to beets library (which I confirmed using the command `beet ls -f "bpm is $bpm" Prove Yourself`. Winamp shows an empty bpm tag in id3v2 and under basic info. 

I thought I may need to update the bpm so I ran `beet update` which suggested the change from 110bpm to 0bpm, which it was able to write successfully (essentially undoing my hard work of tapping enter 3 times to get the bpm). So I guess updating here prefers whatever is currently written in the file over what's in the beets library? 

I then tried modifying the config file to what what you see at the bottom of this post but there was no change in behavior. Am I doing something wrong? 

Running this command in verbose (`-vv`) mode: 

```sh
C:\Users\Jared>beet -vv bpm Prove Yourself
user configuration: C:\Users\Jared\AppData\Roaming\beets\config.yaml
data directory: C:\Users\Jared\AppData\Roaming\beets
plugin paths:
Sending event: pluginload
library database: C:\Users\Jared\AppData\Roaming\beets\library.db
library directory: C:\Users\Jared\Music
Sending event: library_opened
bpm: Press Enter 3 times to the rhythm or Ctrl-D to exit



Sending event: database_change
bpm: Added new bpm 110
Sending event: cli_exit
```

### Setup

* OS: Windows 8.1
* Python version: pip linked to 2.7.9, but cmd line linked to 2.7.10 :: Anaconda 2.3.0 (64-bit), 
* beets version: 1.3.17

My configuration (output of `beet config`) is: 

```yaml
import:
    copy: yes
    move: no
    write: yes

plugins: bpm
bpm:
    max_strokes: 3
    overwrite: yes
```

Thanks for the software by the way.

**Comment by jackwilsdon** [comment]
Have you tried running `beet write` after `beet bpm`?

**Comment by jfunction** [comment]
That works, thanks. Though whatever Windows is using for metadata has been removed on the modified files - ie, the Title, Album etc are not visible in explorer. The id3v2 tags work though.

**Comment by jackwilsdon** [comment]
Take a look at #1698, it seems you need to enable the `id3v23` option.

**Comment by jfunction** [comment]
Works like a charm. Thanks.

I'm wondering if there is a way to automatically get beets to write the bpm after it has been entered? Would I need to modify the plugin?

**Comment by jackwilsdon** [comment]
I may add a configuration option to automatically write.

It seems that it was planned on being implemented and maybe forgotten about? Looking at the `get_bpm` function there is a `write` argument that is never used. I'll take a look at implementing it now :+1:.

**Comment by jfunction** [comment]
Yes, I was looking at https://github.com/beetbox/beets/blob/master/beetsplug/bpm.py#L67 and realised I'd be best off hopping into a debugger to figure out what that's all about. I really can't spare the time until this evening, however.

Please let me know if you manage to make the change :)

Thanks

**Comment by jfunction** [comment]
Ah - just taken a look at your PR - makes more sense now. Thanks.

**Comment by jackwilsdon** [comment]
To use the new `bpm` version (when the PR is merged) you'll need to run the `master` version of beets. Take a look at ["How do I run the latest source version of beets?"](https://beets.readthedocs.io/en/latest/faq.html#run-the-latest-source-version-of-beets).
